src/Factory.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace Friendica;
+
+use Psr\Log\LoggerInterface;
+
+/**
+ * Factories act as an intermediary to avoid direct Entitiy instanciation.
+ *
+ * @see Model
+ * @see Collection
+ */
+abstract class Factory
```

Comment:
A general question: Currently, you're are creating the abstract classes outside of the namespace where they're used, I try to add these classes inside their namespaces, because sometimes we need e.g. interfaces, so I try to "package" them in one namespace Iike `Configuration` or `Session` or `Lock`. Do you prefer it outside of these namespaces? If so, why?

Reply:
To prevent repetition in namespaces. I don't like the `Friendica\Database\Database` name for example.

Reply:
Hmm what about: `Friendica\Repository\Base` for the abstract class? So no repeat and better defined what the class "really" is..

Reply:
I'm not sure what you mean, can you please provide an example of the namespace hierarchy you would prefer for Introduction, Account and all their related classes?

Reply:
Instead of a  `Friendica\Factory` class, we name it `Friendica\Factory\Base` and the other classes will override it like:
`class Mastodon extends Base` which, for me, sounds clear enough and we bundle all relevant classes in one namespace.

Reply:
So we would have 
- `Friendica\Factory\Base`
- `Friendica\Repository\Base`
- `Friendica\Model\Base`
- `Friendica\Collection\Base`
?

Reply:
Yep

Reply:
I don't necessary have a problem with having the same class name in different namespaces, but I'm not sure of the improvement over having an abstract class of the same name of the namespace, so at this point it is more about resistance to change than anything else.

Although I usually don't like having concrete classes in the same folder as the abstract class they are inheriting from.

Reply:
Ah I see, I'm sadly the opposite ^^.. I'd like to have all classes of a namespace inside the namespace-directory to bundle it. I see classes outside of their namespace (even the abstract ones) a little bit "lost" in the upper directory and in some cases, I'd like to have a `README` about what the namespace contains and there I've to referrer outside of the namespace to fully describe the own namespace, which feels a little bit wrong

.. But at least that's - as yours - much more like a personal preference than a concrete "issue" :), so we can have it as is

Reply:
Thanks!